core/modules/layout/layout.admin.inc

```diff
@@ -7,6 +7,70 @@
  */
 
 /**
+ * Output a page to enable or disable layout templates.
+ */
+function layout_settings_page($form, &$form_state) {
+  // Get the list of layout options.
+  $config = config('layout.settings');
+  $available_layouts = $config->get('templates');
+
+  $form['#attached']['js'][] = backdrop_get_path('module', 'layout') . '/js/layout.admin.js';
+  $form['#attached']['css'][] = backdrop_get_path('module', 'layout') . '/css/layout.admin.css';
+
+  $form['markup'] = array(
+    '#markup' => t('Only templates selected here will be enabled on layout settings pages.'),
+  );
+  $form['templates'] = array(
+    '#title' => t('Templates'),
+    '#type' => 'checkboxes',
+    '#default_value' => $available_layouts,
+    '#options' => array(),
+    '#wrapper_attributes' => array('class' => array('clearfix', 'layout-options')),
+  );
+  $form['show_all_admin'] = array(
+    '#title' => t('Show all for admins'),
+    '#description' => t('When enabled, all templates will be shown on layout edit pages for users with \'use all templates\' permissions.'),
+    '#type' => 'checkbox',
+    '#default_value' => 1,
+  );
+  // Get the list of layout options.
+  $all_layout_info = layout_get_layout_info();
+  foreach ($all_layout_info as $layout_name => $layout_info) {
+    $form['templates']['#options'][$layout_name] = theme('layout_option', array('layout_info' => $layout_info));
+  }
+  $form['actions'] = array('#type' => 'actions');
+  $form['actions']['submit'] = array(
+    '#type' => 'submit',
+    '#value' =>t('Save configuration'),
+  );
+  return $form;
+}
+
+/**
+ * Submit handler for layout_settings_page().
+ */
+function layout_settings_page_submit($form, &$form_state) {
+  $templates = array_keys(array_filter($form_state['values']['templates']));
+  $active_templates = array();
+  $all_layouts = layout_get_all_configs('layout');
+  foreach ($all_layouts as $layout) {
+    $active_templates[] = $layout['layout'];
+  }  
+  $excluded_active_templates = array_unique(array_diff($active_templates, $templates));
+  if (!empty($excluded_active_templates)) {
+    $message = t('The following templates are in use but have been disabled: %disabled', array('%disabled' => implode(', ', $excluded_active_templates)));
```

Comment:
...now that I think about it, using the word "disabled" here might confuse people that are used to disabling modules and how that works in that context. Perhaps use the word "hidden" instead?